optuna/artifacts/_download.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+from optuna._experimental import experimental_func
+from optuna.artifacts._protocol import ArtifactStore
+
+
+BUFFER_SIZE = 1024 * 1024  # 1MB
+
+
+@experimental_func("4.0.0")
+def download_artifact(
+    artifact_store: ArtifactStore,
+    artifact_id: str,
+    file_path: str,
+) -> None:
+    """Download an artifact from the artifact store.
+
+    Args:
+        artifact_store:
+            An artifact store.
+        artifact_id:
+            The identifier of the artifact to download.
+        file_path:
+            A path to download the artifact to.
+    """
+    with artifact_store.open_reader(artifact_id) as reader, open(file_path, "wb") as writer:
+        while True:
+            chunk = reader.read(BUFFER_SIZE)
+            if not chunk:
+                break
+            writer.write(chunk)
```

Comment:
Although it is not important to guarantee the file safety here because the downloaded file will not be used by Optuna, I have a concern about the file lock here.
Could we, first of all, check whether the `file_path` already exists at `open`?

Reply:
@nabenabe0928 
Thank you for your comments!
Do you mean that we should check if a file exists at `file_path`, and raise error if it does?

Reply:
I think that is kinder.
But warning or error should be discussed more closely.
At least, `os.makedirs` checks whether the target already exists, for example.
https://docs.python.org/ja/3/library/os.html#os.makedirs

Reply:
I have added the check whether a file exists.